### ISSUE TITLE
Scripts loaded from outside of res are suggested to save in res instead of just saving

### ISSUE THREAD

**Issue by me2beats** [godotengine/godot#42153]
**Godot version:**

v3.2.3.stable

**OS/device including version:**

Windows

**Issue description:**

A script that is loaded from a folder other than `res://` is suggested to be saved in `res://` instead of just saving to loaded path

**Steps to reproduce:**
1) Open a new Godot project
2) Open Script Editor (F3)
3) Open a script that is not in `res://` (with File-Open)
4) Change the script code in the editor [not necessary] and try to close (ctrl+w) or save (ctrl+alt+s) the script
5) As you can see, `Save as` popup appears instead of just saving the script source to path where it is loaded from

The same is for loading a script using `load()` or `Gdscript.new()`.

**Minimal reproduction project:**

**Comment by nathanfranke** [comment]
I don't know if this is a bug, since the "Save Resource As" is always relative to `res://` (try typing `/`, `C:\`, etc. inside)

**Comment by me2beats** [comment]
> I don't know if this is a bug, since the "Save Resource As" is always relative to `res://` (try typing `/`, `C:\`, etc. inside)

Does this mean  all resources have to be saved only in `res://`?
But if so, why `ResourceSaver.save()` allows to save script sources anywhere?

**Comment by nathanfranke** [comment]
True. The fix to this would probably be to support all paths in "save resource as" instead of just res://
